src/System.Management.Automation/engine/ExecutionContext.cs

```diff
@@ -1323,82 +1323,81 @@ internal void RemoveAssembly(string name)
         [SuppressMessage("Microsoft.Reliability", "CA2001:AvoidCallingProblematicMethods", MessageId = "System.Reflection.Assembly.LoadFrom")]
         internal static Assembly LoadAssembly(string name, string filename, out Exception error)
         {
-            // First we try to load the assembly based on the given name
-
-            Assembly loadedAssembly = null;
-            error = null;
-
-            string fixedName = null;
-            if (!String.IsNullOrEmpty(name))
-            {
-                // Remove the '.dll' if it's there...
-                fixedName = name.EndsWith(".dll", StringComparison.OrdinalIgnoreCase)
-                                ? Path.GetFileNameWithoutExtension(name)
-                                : name;
-
-                var assemblyString = Utils.IsPowerShellAssembly(fixedName)
-                                         ? Utils.GetPowerShellAssemblyStrongName(fixedName)
-                                         : fixedName;
-
-                try
-                {
-                    loadedAssembly = Assembly.Load(new AssemblyName(assemblyString));
-                }
-                catch (FileNotFoundException fileNotFound)
-                {
-                    error = fileNotFound;
-                }
-                catch (FileLoadException fileLoadException)
-                {
-                    error = fileLoadException;
-                    // this is a legitimate error on CoreCLR for a newly emited with Add-Type assemblies
-                    // they cannot be loaded by name, but we are only interested in importing them by path
-                }
-                catch (BadImageFormatException badImage)
-                {
-                    error = badImage;
-                    return null;
-                }
-                catch (SecurityException securityException)
-                {
-                    error = securityException;
-                    return null;
-                }
-            }
-
-            if (loadedAssembly != null)
+            // First we try to load the assembly based on the filename
+
+            Assembly loadedAssembly = null;
+            error = null;
+
+            if (!String.IsNullOrEmpty(filename))
+            {
+                error = null;
+
+                try
+                {
+                    loadedAssembly = Assembly.LoadFrom(filename);
+                    return loadedAssembly;
+                }
+                catch (FileNotFoundException fileNotFound)
+                {
+                    error = fileNotFound;

```

Comment:
if we were given a filepath and it doesn't exist, it doesn't seem like we should try to load something that happens to have the same name but different path